simulation-blocks/IntervalPulseGenerator.mon

```diff
@@ -59,23 +59,17 @@ event IntervalPulseGenerator {
 	BlockBase $base;
 	IntervalPulseGenerator_$Parameters $parameters;
 
-	/**All the devices for which block will be listening. Set by reflection by InputHelper.*/
-	sequence<string> devices;
-	
-	/**Flag to identify given device identifier is a group or not. Set by reflection by InputHelper.*/
-	boolean isGroup;
-	
-	/**Flag to identify given device identifier is a broadcast device or not. Set by reflection by InputHelper.*/
-	boolean isBroadcastDevice;
+	/** Input handler for schedule a timer @private */
+	CumulocityInputHandler inputHandler;	
 
 	action $validate(dictionary<string, any> $modelScopeParameters) returns Promise {
-		InputHelper ihelper := InputHelper.forBlock(self, $modelScopeParameters);
-		ihelper.setInput($parameters.deviceId, ManagedObject.getName(), new dictionary<string, any>);
-		return InventoryLookup.lookup($parameters.deviceId).andThen(ihelper.declareInput);
+		
+		CumulocityInputParams c8yInputParams := CumulocityInputParams.create($parameters.deviceId, self, "").withFields(new dictionary<string,any>);
+		return c8yInputParams.declare(inputHandlerCreated);
 	}
 
-	action throwNoDevices(string nameOrId) {
-		throw L10N.getLocalizedException("sample_blk_apamax.analyticsbuilder.samples.DeviceLocationInput_no_devices", [<any> nameOrId]);
+	action inputHandlerCreated(CumulocityInputHandler inputHandler) {
+	    self.inputHandler := inputHandler;
```

Comment:
This is not working. Analytics Builder complains about "" as an event type. Maybe not have this as an input block?

Reply:
Yeah, I don't think this need to be input block.
For un-partitioned models (models using specific device), it should work without any more modifications. 
For partitioned models, you need to create timer in broadcast partition so that it can be triggered for all partitions currently active.
```
TimerParams tp := TimerParams.recurring($parameters.interval).withPartition(new Partition_Broadcast);
any _ := $base.createTimerWith(tp);
```